annotations/views/project_views.py

```diff
@@ -60,6 +60,19 @@ def add_text(self, request, pk=None):
         serializer = ProjectSerializer(project)
         return Response(serializer.data)
 
+    @action(detail=True, methods=['DELETE'], url_name='deletetext')
+    def delete_text(self, request, pk=None):
+        text_id = request.data['text_id']
+
+        submitted = Appellation.objects.filter(occursIn_id=text_id, submitted=True)
+        if submitted:
+            return Response(status=status.HTTP_412_PRECONDITION_FAILED)
+        
+        project = get_object_or_404(TextCollection, pk=pk)
+        project.texts.filter(pk=text_id).delete()
+
+        return Response(status=status.HTTP_200_OK)
```

Comment:
Also lets return a 204 since it is successful and we are not returning anything. See this stack post for more clarification if you are interested. https://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete